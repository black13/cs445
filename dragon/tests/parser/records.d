program example();
type foo = record
    a: integer;
    b: real;
end;
var a: foo;
begin
    a.a := 0;
    a.b := 0
end.
