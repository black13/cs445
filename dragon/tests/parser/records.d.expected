PROGRAM `example` WITH ARGS:
AND VARIABLES:
  DECLARATION WITH TYPE:
    REFERENCE `foo`  OF NAMES:
    `a`
AND SUBPROGRAMS:
DOES:
  ASSIGN
    a.a
  TO
    LIT `0`
  ASSIGN
    a.b
  TO
    LIT `0`
-- done dumping ast --
